### PR TITLE
SM-61 Stream interface

### PR DESCRIPTION
**Pull request by valerio8787**

### CODE REVIEW

File: src/SprykerMiddleware/Service/Process/Model/StreamServiceInterface.php [hunk 1]
```diff
@@ -2,20 +2,32 @@
...
+use SprykerMiddleware\Shared\Process\Stream\StreamInterface;
+
 interface StreamServiceInterface
 {
     /**
-     * @param resource $stream
+     * Specification:
+     * - Read data from given stream.
+     *
+     * @api
```
Comment (PhilinTv) [inline src/SprykerMiddleware/Service/Process/Model/StreamServiceInterface.php]: This interface is not a part of Core API.

File: src/SprykerMiddleware/Service/Process/ProcessServiceInterface.php [hunk 2]
```diff
@@ -2,20 +2,32 @@
...
      *
      * @return array
      */
-    public function readJson($stream);
+    public function readJson(StreamInterface $stream);
 
     /**
-     * @param resource $stream
+     * Specification:
+     * - Write given data to given JSON stream.
```
Comment (PhilinTv) [inline src/SprykerMiddleware/Service/Process/ProcessServiceInterface.php]: Why are you mixing Stream (read/write) and data formatter everywhere?

File: src/SprykerMiddleware/Zed/Process/Business/Pipeline/PipelineInterface.php [hunk 3]
```diff
@@ -22,4 +23,21 @@ public function pipe(callable $operation);
      * @return mixed
      */
     public function process($payload);
+
+    /**
+     * Process the payload.
```
Comment (PhilinTv) [inline src/SprykerMiddleware/Zed/Process/Business/Pipeline/PipelineInterface.php]: Is that Specification?

File: src/SprykerMiddleware/Zed/Process/ProcessDependencyProvider.php [hunk 4]
```diff
@@ -208,33 +225,43 @@ protected function getProcesses(): array
...
 
     /**
-     * @return \SprykerMiddleware\Zed\Process\Dependency\Plugin\Hook\PostProcessorHookPluginInterface[][]
+     * @return \SprykerMiddleware\Zed\Process\Dependency\Plugin\Stream\ProcessStreamPluginInterface[]
      */
-    protected function getPostProcessorHooksStack(): array
+    protected function getStreamPluginsStack()
+    {
+        return [
+            new JsonStreamPlugin(),
```
Comment (PhilinTv) [inline src/SprykerMiddleware/Zed/Process/ProcessDependencyProvider.php]: Should be provided on project level.

**Comment by valerio8787** [comment]
Already merged

**Pull request closed without merging**